python/ray/data/_internal/stats.py

```diff
@@ -88,23 +88,40 @@ class _StatsActor:
     """Actor holding stats for blocks created by LazyBlockList.
 
     This actor is shared across all datasets created in the same cluster.
-    The stats data is small so we don't worry about clean up for now.
+    In order to cap memory usage, we set a max number of stats to keep
+    in the actor. When this limit is exceeded, the stats will be garbage
+    collected in FIFO order.
 
     TODO(ekl) we should consider refactoring LazyBlockList so stats can be
     extracted without using an out-of-band actor."""
 
-    def __init__(self):
+    def __init__(self, max_stats=100 * 1000):
```

Comment:
A bit high?

Reply:
How about 1 or 10k?

Reply:
Reduced to 10k. IIRC there were people talking about number of blocks above 1000, so set it 10k for safety (it won't take significant amount of bytes anyway).

Reply:
Hm it looks uuid is per Dataset not per block, lowering down to 1k.